client/src/Backoffice/components/Main/components/modals/EditRideModal/EditRideModal.tsx

```diff
@@ -0,0 +1,127 @@
+import { Box, Button, Modal, Typography } from '@mui/material';
+import ClearIcon from '@mui/icons-material/Clear';
+import { useState } from 'react';
+import { FormProvider, SubmitHandler, useForm } from 'react-hook-form';
+import { api, getGuestToken } from '../../../../../../Config.ts';
+import { Ride, FetchError } from '../../../../../../api-client';
+import EditRideInfo from './EditRideInfo/EditRideInfo.tsx';
+
+const style = {
+  position: 'absolute' as const,
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 670,
+  bgcolor: 'background.paper',
+  borderRadius: '4px',
+  boxShadow: 24,
+  p: 2
+};
+
+interface EditRideModalProps {
+  open: boolean;
+  handleModal: (shouldOpen: boolean) => void;
+  ride: Ride;
+}
+
+function EditRideModal({ open, handleModal, ride }: EditRideModalProps) {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const methods = useForm<Ride>();
+
+  const { handleSubmit, trigger } = methods;
+
+  const nextStepHandler = async () => {
+    const isStepValid = await trigger([
+      'firstName',
+      'lastName',
+      'origin',
+      'destination',
+      'cellphone',
+      'passengerCount',
+      'comment'
+    ]);
+
+    if (isStepValid) {
+      await handleSubmit(onSubmit)();
+    }
+  };
+
+  const onSubmit: SubmitHandler<Ride> = async (data) => {
+    setIsSubmitting(true);
+    setErrorMessage(null);
+
+    const updatedRide = {
+      ...data,
+      state: ride.state,
+      rideId: ride.rideId,
+      requestTimeStamp: ride.requestTimeStamp
+    };
```

Comment:
כשבדקתי את זה השעה של הנסיעה מתרסטת:
![image](https://github.com/Ezer-Lachaim/EZLA/assets/12518219/38da746b-f13d-403e-b522-16876c516202)

לדעתי אם תוריד את השורה הזאת זה יסתדר:
```suggestion
      rideId: ride.rideId
    };
```